src/faucet/controller.rs

```diff
@@ -217,27 +217,26 @@ impl FaucetController {
             Ok(addr) => {
                 spawn_local(async move {
                     catch_all(faucet.error_messages, async move {
+                        faucet.send_disabled.set(true);
+
                         let rpc = Provider::from_network(network);
                         let from = faucet_address(info)
                             .await
                             .map_err(|e| anyhow::anyhow!("Error getting faucet address: {}", e))?
                             .to_filecoin_address(network)?;
-
-                        faucet.send_disabled.set(true);
                         let nonce = rpc.mpool_get_nonce(from).await?;
-                        let mut msg = message_transfer(from, addr, info.drip_amount().clone());
-                        msg.sequence = nonce;
-                        let msg = rpc.estimate_gas(msg).await?;
-                        match sign_with_secret_key(LotusJson(msg.clone()), info).await {
-                            Ok(LotusJson(smsg)) => {
-                                let cid = rpc.mpool_push(smsg).await?;
+                        let raw_msg = message_transfer(from, addr, info.drip_amount().clone());
+                        let msg = rpc.estimate_gas(raw_msg).await?;
+                        match signed_fil_transfer(LotusJson(msg), nonce, info).await {
```

Comment:
@sudo-shashank, can you please explain how this change mitigates the issue?

Reply:
fixed now by generating the message in backend